android-components/components/support/utils/src/main/java/mozilla/components/support/utils/WebURLFinder.kt

```diff
@@ -96,27 +96,24 @@ class WebURLFinder {
         }
 
         /**
-         * Check if string is a Web URL.
+         * Checks if the given [String] is a valid web URL.
          *
+         * Valid URI schemes: 'http:', 'https:', 'about:', 'data:'.
          *
-         * A Web URL is a URI that is not a `file:` or
-         * `javascript:` scheme.
+         * Invalid URI schemes: 'file:', 'javascript:', 'content:'.
          *
-         * @param string to check.
-         * @return `true` if `string` is a Web URL.
+         * @return True if the [String] is a valid web URL.
          */
         @SuppressWarnings("TooGenericExceptionCaught")
-        fun isWebURL(string: String): Boolean {
-            try {
-                URI(string)
-            } catch (e: Exception) {
-                return false
-            }
-
-            return !(
-                URLUtil.isFileUrl(string.lowercase(Locale.ROOT)) ||
-                    URLUtil.isJavaScriptUrl(string.lowercase(Locale.ROOT))
-                )
+        fun String.isValidWebURL() = try {
+            URI(this)
+
+            val safeUri = lowercase(Locale.ROOT)
+            !(URLUtil.isFileUrl(safeUri) ||
+                URLUtil.isJavaScriptUrl(safeUri) ||
+                URLUtil.isContentUrl(safeUri))
```

Comment:
should all of these checks be inverted/notted?

Reply:
Just for context, nothing around the logic flow has changed here either. Good question, as far as I can tell this is slightly more optimal in that it will fail quicker if it is a file URL or Javascript URL

Reply:
I've pulled this out into a function, hopefully it's a little more user-friendly